### ISSUE TITLE
Method 'Evaluate' in type 'JSPool.JsEngineWithOwnThread' from assembly 'JSPool, Version=2.0.1.0, Culture=neutral, PublicKeyToken=2fc7775f73072640' does not have an implementation.

### ISSUE THREAD

**Issue by richayrich** [Daniel15/JSPool#23]
Hi,

I'm trying to get React working, which when installed through NuGet pulled in JSPool automatically.
I'm seeing this exception when loading a .jsx file:

```
[TypeLoadException: Method 'Evaluate' in type 'JSPool.JsEngineWithOwnThread' from assembly 'JSPool, Version=2.0.1.0, Culture=neutral, PublicKeyToken=2fc7775f73072640' does not have an implementation.]
   JSPool.JsPool.<CreateEngineFactory>b__2_0() +0
   JSPool.JsPool`1.CreateEngine() +19
   JSPool.JsPool`1.PopulateEngines() +18
   JSPool.JsPool`1..ctor(JsPoolConfig`1 config) +191
   React.JavaScriptEngineFactory.CreatePool() +373
   React.JavaScriptEngineFactory..ctor(JsEngineSwitcher jsEngineSwitcher, IReactSiteConfiguration config, IFileSystem fileSystem) +119
   lambda_method(Closure , Object[] ) +128
   React.TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) +682
```

Any ideas? I've tried downgrading to both 2.0.0 and 0.4.1, which result in slightly different errors.

Thanks,
Rich

**Comment by heinzmuller** [comment]
@richayrich Have you found any fix/workaround for this error? Got this error today myself

**Comment by JoachimL** [comment]
We're also struggling with this. Any workarounds?

**Comment by richayrich** [comment]
@heinzmuller  Nope, I haven't progressed past this error.

**Comment by Daniel15** [comment]
Hmm that's very strange. I'll try to take a look tonight and see what's
happening!

Are you on ASP.NET Core, or the "classic" ASP.NET?

**Comment by Daniel15** [comment]
I'm very confused by this error, because `Evaluate` does have an implementation: https://github.com/Daniel15/JSPool/blob/7aea5a2f113dcf8e54b8097e3d8e1c967ae43940/src/JSPool/JsEngineWithOwnThread.cs#L236

**Comment by heinzmuller** [comment]
@Daniel15 Did you manage to replicate it?

**Comment by Daniel15** [comment]
I haven't been able to replicate this. I tested with .NET Framework 4.5.2:
![](http://ss.dan.cx/2017/05/localhost6543test.jsx_-_Google_Chrome_11-21.12.56.png)

And with .NET Core 1.1:
![](http://ss.dan.cx/2017/05/localhost6786test.jsx_-_Google_Chrome_11-21.21.59.png)

Could you please check the file versions of `React.Core.dll`, `JSPool.dll` and `JavaScriptEngineSwitcher.Core.dll` in your site's `bin` directory?

**Comment by Taritsyn** [comment]
This error is caused by changes in the [JavaScript Engine Switcher version 2.4.0](https://github.com/Taritsyn/JavaScriptEngineSwitcher/releases/tag/v2.4.0): 

> In `IJsEngine` interface was added overloaded versions of the `Evaluate`, `Evaluate<T>` and `Execute` methods, which take the document name as second parameter.

You need to roll back the JavaScript Engine Switcher to version 2.3.X.

**Comment by Daniel15** [comment]
Thanks for the info @Taritsyn! Ideally backwards-incompatible changes to the public API (like adding new overloads) should be in a new major version (like `3.0.0`) as per [semver](http://semver.org/#spec-item-8).

I'll bump the JavaScriptEngineSwitcher version in the next JSPool release.

**Comment by Taritsyn** [comment]
@Daniel15 I recommend you inherit wrappers not from the [`IJsEngine`](https://github.com/Taritsyn/JavaScriptEngineSwitcher/blob/master/src/JavaScriptEngineSwitcher.Core/IJsEngine.cs) interface, but from the [`JsEngineBase`](https://github.com/Taritsyn/JavaScriptEngineSwitcher/blob/master/src/JavaScriptEngineSwitcher.Core/JsEngineBase.cs) base class.

**Comment by Taritsyn** [comment]
> Ideally backwards-incompatible changes to the public API (like adding new overloads) should be in a new major version (like `3.0.0`) as per [semver](http://semver.org/#spec-item-8).

Next time I will do so. This time I considered that adding new methods is not a serious API change.

**Comment by Daniel15** [comment]
Should be fixed in JSPool 3.0. I'll publish an updated version of ReactJS.NET that uses JSPool 3.0 shortly.
